### PR TITLE
feat(engine): allow mergify app to be pinged with its url

### PR DESCRIPTION
**Pull request by Greesb**
Fixes MRGFY-1217

### CODE REVIEW

File: mergify_engine/engine/commands_runner.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
 
 LOG = daiquiri.getLogger(__name__)
 
-COMMAND_MATCHER = re.compile(r"^@Mergify(?:|io) (\w*)(.*)", re.IGNORECASE)
+COMMAND_MATCHER = re.compile(r"^(?:https://github.com/|@)Mergify(?:|io) (\w*)(.*)", re.IGNORECASE)
```
Comment (jd) [inline mergify_engine/engine/commands_runner.py:40]: Hum I think `:?` makes the URL prefix or the `@` optional. I'd rather keep them mandatory.

**Review by jd — CHANGES_REQUESTED** [review]

File: mergify_engine/engine/commands_runner.py (hunk 1 above)
Comment (Greesb) [reply mergify_engine/engine/commands_runner.py:40]: No the `:?` is just to indicate that its a non-capturing group

**Review by jd — COMMENTED** [review]
@sileht That looks good to me, but I imagine it won't work with GHES.
Do we want to make that work for GHES too?

We already don't look into the right username though.

File: mergify_engine/engine/commands_runner.py [hunk 2]
```diff
@@ -37,7 +37,10 @@
 
 LOG = daiquiri.getLogger(__name__)
 
-COMMAND_MATCHER = re.compile(r"^@Mergify(?:|io) (\w*)(.*)", re.IGNORECASE)
+COMMAND_MATCHER = re.compile(
+    r"^(?:https://github.com/|@)Mergify(?:|io) (\w*)(.*)",
```
Comment (sileht) [inline mergify_engine/engine/commands_runner.py:41]: We should use config.GITHUB_URL instead of the hardcoded URL to make it work with GHES

**Review by sileht — DISMISSED** [review]

**Comment by sileht** [comment]
> @sileht That looks good to me, but I imagine it won't work with GHES. Do we want to make that work for GHES too?

We have the domain in the configuration

> We already don't look into the right username though.

People use alreadyy @mergify on GHES and this account doesn't exists on them, so... I'm not sure we can do better.

File: mergify_engine/engine/commands_runner.py (hunk 2 above)
Comment (Greesb) [reply mergify_engine/engine/commands_runner.py]: Done @sileht

**Review by sileht — APPROVED** [review]

**Pull request merged**